docs/source/user-guide/dataset_object.rst

```diff
@@ -3,8 +3,139 @@
 ====================
 The Dataset Object
 ====================
-The dataset is one of the basic blocks of deepchecks. It is a container for the data and its relevant metadata, like special column names (label, date, index, etc).
-Some of the checks allows to use a dataframe directly, but some others requires the metadata in order to run, so they are limited to working only with Datasets.
+
+
+A Quick Introduction
+====================
+
+The Dataset Object is a simple and robust Object to store tabular data and metadata relevant to Machine Learning.
+
+Common information used in Machine Learning that is aggregated in the Dataset:
+ - features
+    List of column names. This is the features that are passed to the model. If not defined, columns not defined as something else is considered a feature.
+ - cat_features
+    List of column names. A subset of the features. Categorical features normally require some preprocessing before being passed to the model.
+ - label
```

Comment:
Actually, when reading the doc I think we should rename the label parameter to target.
What do you guys think?
@shir22 @noamzbr @benisraeldan @matanper  @nirhutnik @JKL98ISR